mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp

```diff
@@ -95,6 +95,45 @@ LogicalResult MIOpenEnabled(const Conv2dGenerator::Config& conf) {
   return LogicalResult::success(layoutSupported && noBF16);
 }
 
+LogicalResult validGemmSizes(const Conv2dGenerator &convGen) {
+  const int64_t waveSize = 64;
+  const auto config = convGen.getConfig();
+  if (config.operation != miopen::Conv2DBwdDataOpType)
+    return LogicalResult::success();
+
+  // obtain GemmSize
+  int64_t gemmM, gemmN, gemmK;
+  std::tie(gemmM, gemmN, gemmK) = convGen.getBwdDataGemmSizes(0);
+  if (config.xdlops) {
+    // unsupported xdlops-gemm
+    if (gemmM % 16 != 0 && gemmN % 64 != 0)
+      return LogicalResult::failure();
+
+    if (!((gemmM * gemmN) % 256 == 0 && (gemmK * gemmM) % waveSize == 0 &&
```

Comment:
These are some really strict conditions. Where do they come from?

Reply:
The same test in the lowering pass: https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/miopen-dialect/mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h#L873

Reply:
Then we should share code between that component instead of copy/paste. What if the condition changed in one location and we forget to update the other?

Reply:
I see this PR as a temporary workaround. We'll remove these checks when padding kernels are done. So I wanted to limit the changes within IsApplicable()-related code and not touch other code. For the same reason, I put validGemmSizes() in the mlir-miopen-lib.cpp scope, not conv2dGenerator::isApplicable(), as I don't want to affect other callers to conv2dGenerator::isApplicable().